src/transformations.jl

```diff
@@ -0,0 +1,569 @@
+##################
+## LLA <-> ECEF ##
+##################
+
+immutable LLAfromECEF{Datum} <: AbstractTransformation{LLA, ECEF}
+    a::Float64      # major axis
+    f::Float64      # flattening
+    e2::Float64     # Eccentricity squared
+    e2m::Float64    # 1 - e2m
+    e2a::Float64    # |e2m|
+    e4a::Float64    # e2m^2
+
+    datum::Datum
+
+    function LLAfromECEF(a,f,e2,e2m,e2a,e4a,datum)
+        if !(isfinite(a) && a > 0)
+            error("Major radius is not positive")
+        end
+        if !(isfinite(f) && f < 1)
+            error("Minor radius is not positive")
```

Comment:
Negative f is allowed.  So you could change major and minor to equatorial and polar.  With the ecef conversions, f needs to be in (-inf,1).  For the series approximation, the ellipsoid needs to be close to spherical, i.e., f in (-0.02,0.02), say.